trl/trainer/reward_trainer.py

```diff
@@ -205,6 +225,33 @@ def __init__(
             self.use_reward_data_collator = True
         else:
             self.use_reward_data_collator = False
+
+        if "input_ids" not in train_dataset.column_names:
+            with PartialState().local_main_process_first():
+                fn_kwargs = {"tokenizer": tokenizer}
+                train_dataset = train_dataset.map(
+                    _tokenize,
+                    batched=True,
+                    fn_kwargs=fn_kwargs,
+                    num_proc=args.dataset_num_proc,
+                )
+                train_dataset = train_dataset.filter(
```

Comment:
This filter is important because otherwise you get samples that exceed the model's context length and get truncated => noisy signal because the model can miss the proper label

I could remove it and leave it up to the user's discretion if you prefer.